src/analysis/souffle/examples/simple_may_will/simple_may_will_pass.authlogic

```diff
@@ -0,0 +1,31 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//-------------------------------------------------------------------------------
+
+"EndUser" says {
+    ownsTag("EndUser", "rawVideo").
+    "P1" canSay hasTag(PathX, "rawVideo") :- isAccessPath(PathX).
+    may("P2", "doLocalComputation", "rawVideo").
+}
+
+// It would be better if we had control over the manifest syntax and could just 
+// extend it so we could include:
+//
+// particle P2
+//  
+"P2" says {
+    will("doLocalComputation", "R.P2#1.foo").
```

Comment:
Having `"R.P2#1.foo"` as the access path indicates to me that this `will` claim is attaching itself to a particular particle's access path rather than attaching to all particles that implement this `ParticleSpec`. Is that right? If the `ParticleSpec` represents a particular class of device or a particular version of a piece of software, I would expect that it `will` do the same thing in all of its instantiations.

If what is currently written is what we want, I need to re-work the numbering of particle instances. It's currently not as local to a recipe nor a particular `ParticleSpec` as one might like.

Reply:
I want it to work the way you want it to work. The comment on line 27 was just not finished. I fixed it now. See also the comment in the other file `simple_may_will_fail.authlogic`. The auth logic code is just what should be produced from one specific instantiation of P2 from the recipe.

In general, I think you and I are very much on the same page about how applying annotations to handles in particles should work. So if it seems like I am doing a different thing It's probably either: 1) me making a typo, or 2) me not having a way to hook it up to the way it really works just yet.

Actually, how easy is it to extend the Arcs manifest language or to find some other bit of syntax that we can repurpose for this kind of annotation? (IIRC, we can't change the actual arcs manifest language, and when I looked at the parser for Arcs it was a manually written one rather than one produced by a parser generator).

Reply:
Sorry for reopening this thread.  Where did we end up on this? Ideally, we should just want to be able to specify at the level of particle spec, e.g., `"P2.foo"` instead of `"R.P2#1.foo"`. Is the plan to use "R.P2#1.foo" for now and move to "P2.foo" later?

(This will also be useful for the ownership PR #215.)

Reply:
I want the annotation in the particle spec to be `P2.foo` and then the backend generates `R.P2#1.foo` for this specific instance. To do this we need to change the arcs manifest syntax, though.

Reply:
Sounds good. So, the summary is that it is not possible to do it as is and needs some code changes.
In any case, I don't think we will need to change arcs syntax. We might be able to just add some inference rules in datalog that instantiate the particles themselves.